### PR TITLE
doc: Tar file installation

### PR DESCRIPTION
**Pull request by amyreye**
- New document about building tarball files for ACRN installation.
- Steps are from GSG.

Signed-off-by: Reyes, Amy <amy.reyes@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:103]: This chapter really starts when you untar the files on the target device rootfs, so I would move the section title to just before the "Insert the USB disk you just used into the target..." step.

Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst:103]: This section could still be a different section but I would then call it something like: "Final Configuration steps" or something like that.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:59]:
```suggestion
```

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:98]:
```suggestion
         sudo cp /usr/lib64/acrn/acrn.<board>.<scneario>.bin /boot/acrn/acrn.bin
```

Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst:98]: I would move this line out of this block because this new command does not take the source file from the USB stick (so it's the odd one out in the mix). I would put it in the section below and add a note telling the user he needs to replace `<board>` and `<scenario>` with his own values.

Comment (amyreye) [reply doc/tutorials/tarball_installation.rst:98]: @gvancuts The end of your suggested command is `/boot/acrn/acrn.bin` -- does this rename the bin file?

Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst:98]: Yes, it does. It's on purpose so that it matches the name used in the Grub example below. We can also choose not to rename it but in that case we need to update the Grub section below (`multiboot2` line) to match the name used here.

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: I placed the heading a little higher than your recommendation. Please review.

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: I chose to keep the original name and change the Grub example accordingly. Also, there were two sets of copy commands, one before extracting the tar files and one after extracting the tar files. I consolidated all of the copy commands and put them before extracting the tar files. Please review.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:72]: We can actually remove this, the file is in the tarball that's being generated.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:84]:
```suggestion
```

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:109]:
```suggestion
#. Install the serial configuration tool and configuration file in the target system as follows:
```

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:113]:
```suggestion
      sudo apt install setserial
      sudo cp /usr/lib64/acrn/serial.conf /etc/
```

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [inline doc/tutorials/tarball_installation.rst:69]: This note can now be moved further down when we install the `setserial` tool

**Review by gvancuts — COMMENTED** [review]
Hi @amyreye , your changes look good!

I have made some further suggestions to improve the flow especially regarding the `serial.conf` file. I have also filed a PR to make it even easier to the user so that we would not have to copy it around at all (see PR #7108). If you want, we can wait until this gets merged and then the specific stuff about that `serial.conf` file can all go away completely.

**Comment by amyreye** [comment]
> Hi @amyreye , your changes look good!
> 
> I have made some further suggestions to improve the flow especially regarding the `serial.conf` file. I have also filed a PR to make it even easier to the user so that we would not have to copy it around at all (see PR #7108). If you want, we can wait until this gets merged and then the specific stuff about that `serial.conf` file can all go away completely.

Sounds good. I will wait until the PR is merged.

File: doc/tutorials/tarball_installation.rst
Comment (dbkinder) [inline doc/tutorials/tarball_installation.rst:246]: Should we change this to show Ubuntu 20.04?

**Review by dbkinder — COMMENTED** [review]
Looking good... a few tweeks

File: doc/develop.rst
Comment (dbkinder) [inline doc/develop.rst:64]: since we changed the title not to say tarball, how about changing the file name too

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst:246]: > Should we change this to show Ubuntu 20.04?

Yes, we should

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: @dbkinder Do you have a suggested change, or keep as is?

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: @gvancuts I didn't make this change, because your other comment says that mentions of serial.conf file can all go away completely.

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: @gvancuts I didn't make this change, because your other comment says that mentions of serial.conf file can all go away completely.

File: doc/tutorials/tarball_installation.rst
Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: @gvancuts I didn't make this change, because your other comment says that mentions of serial.conf file can all go away completely. Please check.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst]: That's correct, forget about this suggestion.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst]: Yes, you can drop this suggestion.

File: doc/tutorials/tarball_installation.rst
Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst]: You removed the note completely, didn't you?

The way I understand this you only really need to install the `setserial` tool **if** you have a non-standard serial port. In that case you should install `setserial` that will use the `serial.conf` that we generated and installed. I'm thinking the easiest is to ask the user to install `setserial` regardless, I don't know that there would be any side-effect even in the case you only have standard serial ports configured. @NanlinXie , can someone confirm?

Comment (amyreye) [reply doc/tutorials/tarball_installation.rst]: @gvancuts Yes, I deleted the note.

Comment (gvancuts) [reply doc/tutorials/tarball_installation.rst]: Sounds good, thanks @amyreye !

@NanlinXie , one more question to the engineering team, is there any other action required to get the non-standard serial port configured? Or is installing the `setserial` package and the custom `serial.conf` is sufficient? I believe it is but I do not have first hand experience with this.

Comment (NanlinXie) [reply doc/tutorials/tarball_installation.rst]: @wuxyintel , could you help to reply @gvancuts's question?

**Review by dbkinder — APPROVED** [review]

File: doc/tutorials/tarball_installation.rst
Comment (wuxyintel) [reply doc/tutorials/tarball_installation.rst]: @gvancuts @NanlinXie for acrn v2.7, non-standard ports are emulated by hypervisor only when user config them in scenario file manually, I think these non-standard ports in scenario file will be configured in scenario file by default in acrn v2.8.
If non-standard ports are emulated by hypervisor, serial.conf will be generated during hypervisor build-time. Installing the setserial package and the custom serial.conf in service VM is sufficient. Thanks!

**Pull request merged**
